src/envalid.d.ts

```diff
@@ -101,10 +99,15 @@ export function cleanEnv<T>(
  * @param options An object that specifies options for cleanEnv.
  */
 export function cleanEnv<T>(
-    environment: NodeJS.ProcessEnv,
+    environment: unknown,
     validators?: { [K in keyof T]: ValidatorSpec<T[K]> },
     options?: CleanOptions
-): Readonly<T> & CleanEnv
+): Readonly<T> & CleanEnv & { readonly [varName: string]: string }
```

Comment:
I think introducing `{ readonly [varName: string]: string }` will cause all keys on return value definitely defined, which is not suitable for the situation.

For example, `process.env.SOME_KEY` may be undefined, and should be cautiously tested against `undefined`, but this line means there will always be `SOME_KEY` on return value.

Reply:
Did you read the helpful explanatory note just beneath this line? :)  This is the correct return type for the way this function behaves, but there's an overload just above that does what you're expecting.

This overload of `cleanEnv` only comes into effect if it's invoked _without_ the option `strict: true` -- as [noted in the readme](https://github.com/af/envalid/blob/master/README.md#strict-mode), strict mode ensures that _only_ the fields passed in as validators to `cleanEnv` are returned in the output. Otherwise other values may be present, and we don't want to prevent accessing them.

If you do invoke `cleanEnv` with `strict: true` (and I strongly recommend it!) the return value of the [overload just above this one](https://github.com/af/envalid/blob/f100675244c81236f4d8d905b4eebc916f7d145e/src/envalid.d.ts#L90-L94) will be what you get. This return type indeed does not carry the risk of accessing properties other than the ones you validated.

There's probably another debate to be had for whether `strict` should be the default (I think there's a pretty good case to be made for it!) but that would represent a breaking change for this library and I think it's out of the scope of this PR.

Reply:
We could probably change
```ts
Readonly<T> & CleanEnv & { readonly [varName: string]: string }
```
to be 

```ts
Readonly<T> & CleanEnv & { readonly [varName: string]: string | undefined }
```
instead, though, so they won't be definitely defined. Does that sound preferable?

Reply:
I think that one is preferable, which reduces possible `undefined` in runtime, and that's (IMO) why type system was introduced into TypeScript.

I suggests that never assume a property is always present or of the type you wanted, unless there are assurances of that. In this case, when using non-strict ones, you cannot ensure any key of type string is always present on the return value.

Thanks for your work!